### PR TITLE
Verify Twitter Credentials

### PR DESCRIPTION
**Pull request by stulees**
## Description
Calling Twitter Service _verify-credentials_ endpoint when showing Twitter component instance with the company id of presentation

Handling OK response by checking `success` property and updating UI bound values accordingly to ensure to show appropriate UI

Applying spinner loader to show/block when processing requests to Twitter Service and Oauth

## Motivation and Context
Twitter development

## How Has This Been Tested?
First validated Connect button only showed in https://apps-stage-8.risevision.com/templates/edit/2887af36-4533-4325-8ad6-2f2b943072c4/?cid=cb73b359-b827-4e38-9893-9ee28d8e5676

Then authenticated with my Twitter account via Twitter Widget in https://apps-stage-8.risevision.com/editor/workspace/2c5efad7-d6f9-4d24-8441-c4a1dc86f3fe?cid=cb73b359-b827-4e38-9893-9ee28d8e5676

Then back to first presentation and validated nothing shown (credentials verified) in Twitter UI. 

## Release Plan:
- As the Submitter, upon requesting review of this pull request, I confirm that the [Release Checklist](https://help.risevision.com/hc/en-us/articles/360031119991) has been completed. 
- As the Reviewer, upon approving the changes in this PR, I confirm I have reviewed and I agree that the [Release Checklist](https://help.risevision.com/hc/en-us/articles/360031119991) has been completed

#### Release Checklist Items Skipped?
n/a

### CODE REVIEW

File: web/scripts/template-editor/components/directives/dtv-component-twitter.js [hunk 1]
```diff
@@ -32,9 +47,31 @@ angular.module('risevision.template-editor.directives')
             show: function () {
               element.show();
               $scope.componentId = $scope.factory.selected.id;
+              _validateCredentials();
             }
           });
 
+          function handleConnectionFailure() {
```
Comment (santiagonoguez) [inline web/scripts/template-editor/components/directives/dtv-component-twitter.js]: @stulees minor, this private function should start with '_', right ?

Comment (stulees) [reply web/scripts/template-editor/components/directives/dtv-component-twitter.js]: @santiagonoguez Yes it should, change made, thanks!

**Comment by stulees** [comment]
@santiagonoguez Yes all good regarding authenticating with our Twitter settings now. I was curious to do it via Twitter widget as I haven't done it in a long time and was nice to see our Twitter settings verified correctly afterwards, which Shea was hoping for so our existing users don't have to re-authenticate.

File: test/unit/template-editor/components/services/svc-twitter-credentials-validation.js [hunk 2]
```diff
@@ -0,0 +1,71 @@
+'use strict';
```
Comment (alex-deaconu) [inline test/unit/template-editor/components/services/svc-twitter-credentials-validation.js]: @stulees just a heads up, these tests are not being run. The file is named incorrectly and from what I can tell is not being included in the unit test task 🤷‍♂. Thanks.

Comment (stulees) [reply test/unit/template-editor/components/services/svc-twitter-credentials-validation.js]: @alex-deaconu Oops, the file name is missing `.tests` in it. I know they pass as I was force targeting that test file locally. I'll fix this up now, thanks for letting me know!

Comment (alex-deaconu) [reply test/unit/template-editor/components/services/svc-twitter-credentials-validation.js]: @stulees 👍

**Pull request merged**